src/proc/mod.rs

```diff
@@ -110,6 +118,45 @@ impl super::TypeInner {
     }
 }
 
+impl crate::TypeInner {
```

Comment:
can we move all the indexable logic into `src/proc/index.rs` and re-export here?

Reply:
Sure.